lib/jxl/enc_patch_dictionary.cc

```diff
@@ -302,6 +303,44 @@ std::vector<PatchInfo> FindTextLikePatches(
     return {};
   }
 
+  // Assumes bitdepths of at most 16 bits.
+  auto color_id = [](float r, float g, float b) {
+    uint64_t rh = std::rint(r * (1 << 16));
+    uint64_t gh = std::rint(g * (1 << 16));
+    uint64_t bh = std::rint(b * (1 << 16));
+    return (rh << 40) | (gh << 20) | bh;
+  };
+  float num_colors_log = 0;
+  float pos_log = 0;
+  // How good a patch has to be to be considered at all.
+  constexpr float kPosFactor = 1.0f;
+  // How good a patch has to be, considering its total number of occurrences, to
+  // be used.
+  constexpr float kPosFactorOccurrences = 64.0f;
+  // How good a patch has to be to override the minimum count of occurrences.
+  // TODO(veluca): think if enabling this is a good idea.
+  float kPosFactorOccurrencesOverride = 1e10f;
```

Comment:
Maybe drop the `kPosFactorOccurrencesOverride` since it's not really used.

Doing patches that are only used once will require a different approach imo, and I don't think it makes sense in lossless mode. In lossy mode it can make sense but it will require a different kind of logic (something that estimates how likely it is that modular encoding + patch signaling overhead is better than vardct encoding).